code/modules/mob/living/carbon/give.dm

```diff
@@ -1,52 +1,184 @@
+		return
+	if(istype(client.click_intercept, /datum/click_intercept/give))
+		give_icon.icon_state = "act_give_off"
+		QDEL_NULL(client.click_intercept)
+		return
+	var/obj/item/I = get_active_hand()
+	if(!I)
+		to_chat(src, "<span class='warning'>You don't have anything in your hand to give!</span>")
+		return
+	if(I.flags & NODROP)
+		to_chat(src, "<span class='warning'>[I] is stuck to your hand, you can't give it away!</span>")
+		return
+	if(I.flags & ABSTRACT)
+		to_chat(src, "<span class='warning'>That's not exactly something you can give.</span>")
+		return
+
+	give_icon.icon_state = "act_give_on"
+	new /datum/click_intercept/give(client)
+
+/**
+ * # Offering Item status effect
+ *
+ * Status effect given to mobs after they've offered an item to another player using the Give Item action ([/datum/click_intercept/give]).
+ */
+/datum/status_effect/offering_item
+	id = "offering item"
+	duration = 10 SECONDS
+	alert_type = /obj/screen/alert/status_effect/offering_item
 
-	if(!iscarbon(target)) //something is bypassing the give arguments, no clue what, adding a sanity check JIC
-		to_chat(usr, "<span class='danger'>Wait a second... \the [target] HAS NO HANDS! AHH!</span>")//cheesy messages ftw
+/obj/screen/alert/status_effect/offering_item
+	name = "Offering Item"
+	desc = "You're currently offering an item someone. Make sure to keep the item in your hand so they can accept it!"
+	icon_state = "offering_item"
+
+/**
+ * # Give click intercept
+ *
+ * While a mob has this intercept, left clicking on a carbon mob will attempt to offer their currently held item to that mob.
+ */
+/datum/click_intercept/give
+	/// If the intercept user has succesfully offered the item to another player.
+	var/item_offered = FALSE
+
+/datum/click_intercept/give/New(client/C)
+	..()
+	holder.mouse_pointer_icon = file("icons/mouse_icons/give_item.dmi")
+	to_chat(holder, "<span class='info'>You can now left click on someone to give them your held item.</span>")
+	RegisterSignal(holder.mob.get_active_hand(), list(COMSIG_PARENT_QDELETING, COMSIG_ITEM_EQUIPPED, COMSIG_ITEM_DROPPED), /datum/proc/signal_qdel)
+	RegisterSignal(holder.mob, list(SIGNAL_ADDTRAIT(TRAIT_HANDS_BLOCKED), COMSIG_CARBON_SWAP_HANDS), /datum/proc/signal_qdel)
+
+
+/datum/click_intercept/give/Destroy(force = FALSE, ...)
+	holder.mouse_pointer_icon = initial(holder.mouse_pointer_icon)
+	holder.mob.give_icon.icon_state = "act_give_off"
+	if(!item_offered)
+		to_chat(holder.mob, "<span class='info'>You're no longer trying to give someone your held item.</span>")
+	UnregisterSignal(holder.mob.get_active_hand(), list(COMSIG_PARENT_QDELETING, COMSIG_ITEM_EQUIPPED, COMSIG_ITEM_DROPPED))
+	UnregisterSignal(holder.mob, list(SIGNAL_ADDTRAIT(TRAIT_HANDS_BLOCKED), COMSIG_CARBON_SWAP_HANDS))
+	return ..()
+
+
+/datum/click_intercept/give/InterceptClickOn(mob/user, params, atom/object)
+	if(user == object || !iscarbon(object))
+		return
+	var/mob/living/carbon/receiver = object
+	if(receiver.stat != CONSCIOUS)
```

Comment:
Probably won't happen... but this can be used to meta check against `TRAIT_FAKEDEATH`

Reply:
Right, I'll have to account for that.

Reply:
Alright so apparently when `TRAIT_FAKEDEATH` is added, they do eventually have their stat set to `UNCONSCIOUS` through a series of signal calls.

Reply:
Won't this still say that they're unconscious then, through the stat check in the message?

Reply:
Yes.